Microsoft.Toolkit.Uwp.Input.GazeControls/GazeFilePicker.xaml.cs

```diff
@@ -0,0 +1,390 @@
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Input.GazeInteraction;
+using Windows.Storage;
+using Windows.Storage.FileProperties;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.Toolkit.Uwp.Input.GazeControls
+{
+    /// <summary>
+    /// Provides file picker dialogs optimized for gaze input
+    /// </summary>
+    public sealed partial class GazeFilePicker : ContentDialog, INotifyPropertyChanged
+    {
+        private Grid _commandSpaceGrid;
+        private Button _newFolderButton;
+        private Button _enterFilenameButton;
+        private Button _selectButton;
+        private Button _cancelButton;
+        private DispatcherTimer _initializationTimer;
+
+        private bool _newFolderMode;
+
+        private ObservableCollection<StorageItem> _currentFolderItems;
+
+        private StorageItem _curSelectedItem;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this is FileSave dialog or a FileOpen dialog
+        /// </summary>
+        public bool SaveMode { get; set; }
+
+        /// <summary>
+        /// Gets the currently selected file in the dialog as a StorageFile
+        /// </summary>
+        public StorageFile SelectedItem { get; private set; }
+
+        private StorageFolder _currentFolder;
+
+        /// <summary>
+        /// Gets or sets the current folder for the file picker dialog
+        /// </summary>
+        public StorageFolder CurrentFolder
+        {
+            get
+            {
+                return _currentFolder;
+            }
+
+            set
+            {
+                RefreshContents(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the list of storage folders that appear as shortcuts
+        /// on top of the folder view
+        /// </summary>
+        public List<StorageFolder> Favorites { get; set; }
+
+        /// <summary>
+        /// Gets or sets the collection of file types that the file open picker displays.
+        /// </summary>
+        public List<string> FileTypeFilter { get; set; }
+
+        /// <inheritdoc/>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GazeFilePicker"/> class.
+        /// </summary>
+        public GazeFilePicker()
+        {
+            this.InitializeComponent();
+
+            this.Opened += OnGazeFilePickerOpened;
+
+            FileTypeFilter = new List<string>();
+
+            _initializationTimer = new DispatcherTimer();
```

Comment:
Make sure you're using DispatcherQueue and DispatcherQueueTimer. `Dispatcher` and `DispatcherTimer` are deprecated.

Reply:
Done.